modules/instance_template/main.tf

```diff
@@ -44,11 +44,18 @@ locals {
 
   all_disks = concat(local.boot_disk, var.additional_disks)
 
-  # NOTE: Even if all the shielded_instance_config values are false, if the
-  # config block exists and an unsupported image is chosen, the apply will fail
-  # so we use a single-value array with the default value to initialize the block
-  # only if it is enabled.
-  shielded_vm_configs = var.enable_shielded_vm ? [true] : []
+  # NOTE: Even if all the shielded_instance_config or confidential_instance_config
+  # values are false, if the config block exists and an unsupported image is chosen,
+  # the apply will fail so we use a single-value array with the default value to
+  # initialize the block only if it is enabled.
+  shielded_vm_configs          = var.enable_shielded_vm ? [true] : []
+  confidential_instance_config = var.enable_confidential_vm ? [true] : []
+
+  on_host_maintenance = (
+    var.preemptible || var.enable_confidential_vm
+    ? "TERMINATE"
```

Comment:
i think the logic here is suspect...

if the user intended to create a "regular" VM (i.,e non preemptable or non-confidential),  *and* they want  the instance to`TERMINATE`, this logic will reset the user's intent to`MIGRATE`?

Reply:
I don't think it's a logic issue so much as a feature request. We don't allow module users to change the host maintenance strategy and instead force *all* regular VM users to choose the `MIGRATE` strategy.

Reply:
you mean surface another variable that denotes the users choice for `scheduling.on_host_maintenance`?
https://cloud.google.com/compute/docs/instances/setting-instance-scheduling-options

where default is "MIGRATE"
```
The default maintenance behavior for instances is to live migrate, 
but you can change the behavior to stop your instance during maintenance events instead.
```

then something the variablewould be like

```hcl
  on_host_maintenance = (
    var.preemptible || var.enable_confidential_vm
    ? "TERMINATE"
    : var.on_host_maintenance
  )
```

(or something like that)

Reply:
Exactly, something like that.

Reply:
ok aded in the variable `on_host_maintenance`.  unfortuantely, its a flat variable like the other onces for scheduling....ideally it would be nested inside an a `scheduling` object bit its a bit late for that (or a separate FR)